server/controllers/MiscController.js

```diff
@@ -717,5 +717,95 @@ class MiscController {
     const stats = await adminStats.getStatsForYear(year)
     res.json(stats)
   }
+
+  /**
+   * GET: /api/custom-metadata-providers
+   *
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   */
+  async getCustomMetadataProviders(req, res) {
+    const providers = await Database.customMetadataProviderModel.findAll()
+
+    res.json({
+      providers: providers.map((p) => p.toUserJson()),
+    })
+  }
+
+  /**
+   * GET: /api/custom-metadata-providers/admin
+   *
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   */
+  async getAdminCustomMetadataProviders(req, res) {
+    if (!req.user.isAdminOrUp) {
+      Logger.error(`[MiscController] Non-admin user "${req.user.username}" attempted to get admin custom metadata providers`)
+      return res.sendStatus(403)
+    }
+
+    const providers = await Database.customMetadataProviderModel.findAll()
+
+    res.json({
+      providers,
+    })
+  }
+
+  /**
+   * PATCH: /api/custom-metadata-providers/admin
+   *
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   */
+  async addCustomMetadataProviders(req, res) {
+    if (!req.user.isAdminOrUp) {
+      Logger.error(`[MiscController] Non-admin user "${req.user.username}" attempted to add admin custom metadata providers`)
+      return res.sendStatus(403)
+    }
+
+    const { name, url, apiKey } = req.body;
```

Comment:
```suggestion
    const { name, url, apiKey } = req.body
```
The repo is not using semicolons and not using trailing commas. I have this rule setup for vscode but still working on improvements for this #2183